src/sentry/integrations/slack/integration.py

```diff
@@ -217,12 +217,28 @@ class SlackReAuthIntro(PipelineView):
     """
 
     def dispatch(self, request, pipeline):
-        if "integration_id" in request.GET:
-            pipeline.bind_state("integration_id", request.GET["integration_id"])
+        integration_id = request.GET.get("integration_id")
+
+        if integration_id:
+            pipeline.bind_state("integration_id", integration_id)
             pipeline.bind_state("user_id", request.user.id)
 
             try:
-                all_channels = _get_channels_from_rules(pipeline)
+                integration = Integration.objects.get(id=integration_id, provider="slack",)
+            except Integration.DoesNotExist:
+                return pipeline.error(IntegrationError("Could not find Slack integration."))
+
+            # We check if there are any other orgs tied to the integration to let the
+            # user know those organizations will be affected by the migration
+            extra_orgs = {}
+            for org in integration.organizations.all():
+                if org != pipeline.organization:
+                    extra_orgs[org.id] = org.slug
```

Comment:
@MeredithAnya I think it might be cleaner to use a filter + map here. Filter out the org for this integration then map it to the slug. We can store the slugs instead of `extra_org_ids` when we do the audit log.

Reply:
Since the `AuditLogEntry` expects the organization instance to be passed, I guess we can look up the org using the slug, since it's unique. It's technically possible that the slug could change but probs very unlikely during the same time they are updating an integration